ui/components/src/JSONEditor.tsx

```diff
@@ -33,24 +39,29 @@ export function JSONEditor<T>(props: JSONEditorProps<T>) {
   }, [props.value]);
 
   return (
-    <CodeMirror
-      style={{ border: `1px solid ${theme.palette.divider}` }}
-      theme={isDarkMode ? 'dark' : 'light'}
-      extensions={[json(), linter(jsonParseLinter()), lintGutter()]}
-      value={value}
-      onChange={(newValue) => {
-        setValue(newValue);
-      }}
-      onBlur={() => {
-        try {
-          const json = JSON.parse(value ?? '{}');
-          if (props.onChange !== undefined) {
-            props.onChange(json);
+    <Box maxWidth={maxWidth} width={width}>
```

Comment:
The `Box` is not needed, you can just pass these props directly to CodeMirror.

It might be even better to have JSONEditor extend CodeMirror and only override the properties that make it a json specific.

Rough idea
```tsx
type JSONEditorProps = Omit<CodeMirrorProps, 'onBlur' | 'theme' | 'extensions'>

function JSONEditor(props: JSONEditorProps) {
  return (
     <CodeMirror 
         {...props}
         extensions={[json(), linter(jsonParseLinter()), lintGutter()]}
      />
}
```

Reply:
ah yeah that's definitely better like that. Thanks @saminzadeh !